dataprofiler/tests/space_time_analysis/dataset_generation.py

```diff
@@ -0,0 +1,256 @@
+                       num_rows: int=1) -> np.array:
+    """
+    Randomly generates an array of categorical chosen out of categories
+
+    :param rng: the np rng object used to generate random values
+    :type rng: numpy Generator
+    :param categories: a list of values that are allowed in a categorical or None
+    :type categories: string, None
+    :param num_rows: the number of rows in np array generated
+    :type num_rows: int
+
+    :return: np array of categories
+    """
+    if categories is None:
+        categories = ["A", "B", "C", "D", "E"]
+
+    return rng.choice(categories, (num_rows,))
+
+
+def get_ordered_column(start: int=0, num_rows: int=1, **kwarg):
+    """
+    Generates an array of ordered integers
+
+    :param start: integer that the ordered list should start at
+    :type str_len_min: int
+    :param num_rows: the number of rows in np array generated
+    :type num_rows: int
+
+    :return: np array of ordered integers
+    """
+    return np.array(list(range(start, num_rows)))
+
+
+def generate_dataset_by_class(rng: Generator, classes_to_generate: List[str]=None,
+                              dataset_length: int=100000,
+                              path: str=None) -> pd.DataFrame:
+    """
+    Randomly a dataset with a mixture of different data classes
+
+    :param rng: the np rng object used to generate random values
+    :type rng: numpy Generator
+    :param classes_to_generate: Classes of data to be included in the dataset
+    :type classes_to_generate: List[str] or None
+    :param dataset_length: length of the dataset generated
+    :type dataset_length: int
+    :param path: path to output a csv of the dataframe generated
+    :type path: str
+
+    :return: pandas DataFrame
+    """
+    possible_classes = ["text", "string", "categorical", "integer", "float",
+                        "ordered", "datetime"]
+    if classes_to_generate is None:
+        classes_to_generate = possible_classes
+
+    dataset = []
+    if "integer" in classes_to_generate:
+        dataset.append(random_integers(rng, num_rows=dataset_length))
+    if "float" in classes_to_generate:
+        dataset.append(random_integers(rng, num_rows=dataset_length))
+    if "datetime" in classes_to_generate:
+        dataset.append(random_datetimes(
+            rng,
+            date_format_list=dp.profilers.DateTimeColumn._date_formats,
+            num_rows=dataset_length
+        ))
+    if "string" in classes_to_generate:
+        dataset.append(random_string(rng, num_rows=dataset_length))
+    if "categorical" in classes_to_generate:
+        dataset.append(random_categorical(rng, num_rows=dataset_length))
+    if "text" in classes_to_generate:
+        dataset.append(random_string(rng, num_rows=dataset_length,
+                                     str_len_min=256, str_len_max=1000))
+    if "ordered" in classes_to_generate:
+        dataset.append(get_ordered_column(num_rows=dataset_length))
```

Comment:
Right now, this doesn't allow it to be modifiable via options. I think we could also refactor with a dict. This might also allow us to better parallelize this result.

A format more similar to:
```python
generate_dataset_by_class(columns_to_generate: List[dict], rng: Generator=None,
                              dataset_length: int=100000,
                              path: str=None)
.
.
.
gen_funcs = {
        'int': random_int,
        'float': random_float,
        'categorical':, random_categorical,
        'ordered':, get_ordered_column,
        'text':, random_text,
        'datetime':, random_datetime,
         ...
    }

# default classes
if columns_to_generate is None:
   columns_to_generate = [...]

dataset = []
for col in columns_to_generate:
    col = copy.deepcopy(col)
    col_gen_name = columns_to_generate.pop("name")
    if col_gen_name not in gen_funcs:
        print(f"generator: {col_gen_name} is not a valid generator.")

    col_generator = gen_funcs.get(col_gen_name)
    dataset.append(col_generator(rng, **col, num_rows=dataset_length)


.
.
.




# data looks like
cols_to_gen = [
    dict(name='datetime', min=,max=,), 
    dict(name='int', min=,max=,), 
    dict(name='float', min=,max=,), 
    dict(name='int', min=,max=,)', 
    dict(name='text', max_len=255), 
    ...]
```


For something like random_text, we could just create a surrogate func:
```python
def random_text(rng: Generator, chars:str=None, num_rows: int=1,
                  str_len_min: int=256, str_len_max: int=1000) -> np.array:
    if str_len_min < 256:
        raise ...
    return  random_string(rng, strings, num_rows, str_len_min, str_len_max)
```

Reply:
`print(f"generator: {col_gen_name} is not a valid generator.")` -> exception

Reply:
`dict(name='datetime', min=,max=,),` -> `dict(gen='datetime', min=,max=,),` or `generator`